### DISCUSSION TITLE
String Encoding

### DISCUSSION THREAD

**Discussion by seekeramento** [pest-parser/pest#703] [Q&A] [answered]
When I need to define a string term, I can do something like this:
char = @{ ASCII_ALPHANUMERIC * }

However, this only works with Latin characters. Any other languages will cause parser error. What are the right alternative to use in my case to define a string that works for different encoding?

**Comment by tomtau** [comment, marked as answer]
https://pest.rs/book/grammars/built-ins.html#unicode-rules

**Reply by seekeramento** [reply]
Thanks. 
For anyone who's facing the same issue like me, please make sure that you updated your pest-parser to the latest. The list of encoding that you can select from are in the abovementioned link. Below are the common options all together for simplicity. This is instead of using  ASCII_ALPHANUMERIC with special characters as used to in my case before:

char = {
ASCII_ALPHANUMERIC | ALPHABETIC | BIDI_CONTROL |
ADLAM |AHOM |ANATOLIAN_HIEROGLYPHS |ARABIC |ARMENIAN |AVESTAN |BALINESE |BAMUM |BASSA_VAH |BATAK |BENGALI |BHAIKSUKI |BOPOMOFO |BRAHMI |BRAILLE |BUGINESE |
BUHID |CANADIAN_ABORIGINAL |CARIAN |CAUCASIAN_ALBANIAN | CHAKMA |CHAM |CHEROKEE |CHORASMIAN |COPTIC |CUNEIFORM | CYPRIOT |CYPRO_MINOAN |
CYRILLIC |DESERET |DEVANAGARI |DIVES_AKURU |DOGRA |DUPLOYAN |EGYPTIAN_HIEROGLYPHS |ELBASAN |ELYMAIC |ETHIOPIC |GEORGIAN |GLAGOLITIC |GOTHIC |
GRANTHA |GREEK |GUJARATI |GUNJALA_GONDI |GURMUKHI |HAN |HANGUL |HANIFI_ROHINGYA |HANUNOO |HATRAN |HEBREW |HIRAGANA |IMPERIAL_ARAMAIC |INHERITED |
INSCRIPTIONAL_PAHLAVI |INSCRIPTIONAL_PARTHIAN |JAVANESE |KAITHI |KANNADA |KATAKANA |KAWI |KAYAH_LI |KHAROSHTHI |KHITAN_SMALL_SCRIPT |KHMER |KHOJKI |KHUDAWADI |LAO |LATIN |LEPCHA |LIMBU |LINEAR_A |LINEAR_B |LISU |LYCIAN |LYDIAN |MAHAJANI |MAKASAR |MALAYALAM |MANDAIC |MANICHAEAN |MARCHEN |MASARAM_GONDI |MEDEFAIDRIN |MEETEI_MAYEK |MENDE_KIKAKUI |MEROITIC_CURSIVE |MEROITIC_HIEROGLYPHS |MIAO |MODI |MONGOLIAN |MRO |MULTANI |MYANMAR |NABATAEAN |NAG_MUNDARI |NANDINAGARI |NEW_TAI_LUE |NEWA |NKO |NUSHU |NYIAKENG_PUACHUE_HMONG |OGHAM |OL_CHIKI |OLD_HUNGARIAN |OLD_ITALIC |OLD_NORTH_ARABIAN |OLD_PERMIC |OLD_PERSIAN |OLD_SOGDIAN |OLD_SOUTH_ARABIAN |OLD_TURKIC |OLD_UYGHUR |ORIYA |OSAGE |OSMANYA |PAHAWH_HMONG |PALMYRENE |PAU_CIN_HAU |PHAGS_PA |PHOENICIAN |PSALTER_PAHLAVI |REJANG |RUNIC |SAMARITAN |SAURASHTRA |SHARADA |SHAVIAN |SIDDHAM |SIGNWRITING |SINHALA |SOGDIAN |SORA_SOMPENG |SOYOMBO |SUNDANESE |SYLOTI_NAGRI |SYRIAC |TAGALOG |TAGBANWA |TAI_LE |TAI_THAM |TAI_VIET |TAKRI |TAMIL |TANGSA |TANGUT |TELUGU |THAANA |THAI |TIBETAN |TIFINAGH |TIRHUTA |TOTO |UGARITIC |VAI |VITHKUQI |WANCHO |WARANG_CITI |YEZIDI |YI |ZANABAZAR_SQUARE |
}

**Reply by NoahTheDuke** [reply]
Maybe we should bundle those together, `UNICODE_ALPHANUMERIC` or similar.

**Reply by seekeramento** [reply]
Cool. This is important to mention. Many thanks. 
I was trying here to show the different possible combinations.
